### DISCUSSION TITLE
Bank of many small bandwidth SDRs to create a large bandwidth capture device

### DISCUSSION THREAD

**Discussion by pfeatherstone** [jgaeddert/liquid-dsp#317] [General]
Suppose I need to capture signals with extremely large bandwidths, lets say with a bandwidth of 1GHz.
You would think you need an SDR or sensor with an instantaneous bandwidth of at least 1GHz.
How about you use a series of SDRs with smaller bandwidths and combine them together using a polyphase synthesis channelizer.

For example, if you had 10 SDRs which have an instantaneous bandwidth of 100MHz, you could off-tune them such that they are equally spaced over the 1GHz bandwidth, synchronize them with a 10MHz external reference, capture everything simultaneously and use something like `firpfbch_crcf_create(LIQUID_SYNTHESIZER, 10, 64, h)` to combine everything together into a 1GHz bandwidth signal.

Would this work? Can you think of any gotchas?
Capturing everything exactly at the same time could be difficult.
You would need to ensure that the gain levels are the same on all channels, AGC is turned off, things like that.

Is it the case that mathematically this holds up, but in practice, it's impossible to get that level of synchronisation across all channels.

**Comment by seonlee2013** [comment]
好啊，邮件收到啦！
Hi, your email is arrived safely

**Comment by pfeatherstone** [comment]
@jgaeddert Do you have any thoughts?

**Reply by jgaeddert** [reply]
I do! I have actually done this on a project once; however the bandwidths were nowhere close to 1 GHz. In practice, you not only need the sample timing to be perfectly aligned, but the sample phase as well. What we did on this program was allow the sampled signals of the various ADCs to overlap, align the streams using the overlapping region, and then re-assemble the components together.

In your example, if you had two radios sampling from the same antenna at 100 MHz sample rates each, but overlapping by 10 MHz, you can use the overlapping region for timing and phase alignment. You can do this by channelizing each of the 100 MHz radios into 10 channels each at 10 MHz, using the overlapping region for alignment, and then reconstructing the 190 MHz wide-band using a synthesizer.

We used the [over-sampled channelizer](https://liquidsdr.org/api/firpfbch2_crcf/) for near-perfect reconstruction. It worked like a charm, but you need a fairly capable front-end, and something in the overlapping band for this to work well.

**Reply by pfeatherstone** [reply]
Thank you! How did you do the alignment? What do you mean by "something" in the overlapping region? What do you mean by capable front-end ?

**Reply by pfeatherstone** [reply]
Does the RF front-end of each radio need to be tailored or be higher spec?
Did you have to transmit something in the overlapping bands simply to do time and phase alignment?

**Reply by jgaeddert** [reply]
Ideally you would have something like a phased array with coherent reference clocks. If you imaging that in the above example you have to radios sampling at the same 100 MHz (split from a single antenna), combining them would just amplify all the signals in the band because the samples streams are

  1. sampled at the exact same time
  2. mixed using the exact same carrier

When they are tuned off each other by e.g. 90 MHz, you can either assume they are carrier locked or use a signal in the overlapping band (something naturally there or something you generate) to ensure coherence.

**Reply by jgaeddert** [reply]
I guess what I'm saying is that if you do have a reference signal that both digitizers can see, it's possible to synchronize the streams without having a common reference clock.
